lib/kitchen/directions/bake_index/v1.rb

```diff
@@ -118,10 +118,25 @@ def bake(book:)
       @metadata_elements = book.metadata.children_to_keep.copy
       @index = Index.new
 
-      book.pages.terms.each do |term_element|
+      chapters = book.chapters
+      book.pages_and_composite_pages.terms.each do |term_element|
         # Markup the term
-        page = term_element.ancestor(:page)
-        term_element.id = "auto_#{page.id}_term#{term_element.count_in(:book)}"
+        begin
+          page = term_element.ancestor(:page)
+        rescue RuntimeError
+          page = term_element.ancestor(:composite_page)
+        end
```

Comment:
[Use exceptions for exceptional behavior only](https://bit.ly/3eBz0Dm), not for normal flow behavior.

I bet you look at this code and think it looks kind of gnarly.  The world is your oyster, so can you envision other approaches that would look better and make you feel better about the code, then see if you can implement them?  I can think of two ways.  **Take a look at them after** you ponder this for yourself a bit: https://gist.github.com/jpslav/7a53c1a4c0a17d2a735f6fd87abb7abc

Reply:
wow thank you mr mentor

Reply:
To show you that I did my homework, my ideas for a different way of doing this were along the lines of "well I could safely inquire about what ancestors are available on an element with `element.ancestors.keys` or use the `has_ancestor?` method, to avoid triggering the exception." I tried to think of a second way for kicks, but I was never any good at not peeking at the upside down text on those brainteaser mysteries as a kid. I think my solution would be more of a bandaid in comparison to your proposed solutions, with all their good refactoring in base Kitchen.

I see you've assigned the second of the two approaches to me in a separate issue. I also see your comment below outlining a two-loop solution. Does it work if I implement your comment below for this PR, and do the yield-multiple-elements refactor later, on a different branch?

Thanks so much for working out these options!